src/utils/modules/tip.ts

```diff
@@ -0,0 +1,491 @@
+import { RANGE_FELT } from '../../global/constants';
+import { logger } from '../../global/logger';
+import type { ProviderInterface } from '../../provider';
+import { BlockTag, type BlockIdentifier, type RPC } from '../../types';
+import type { BlockWithTxs } from '../../types/api';
+import assert from '../assert';
+import { LibraryError } from '../errors';
+
+/**
+ * Result of provider.getTipStatsFromBlocks().
+ * @param {bigint} minTip - minimum tip encountered in the analyzed blocks.
+ * @param {bigint} maxTip - maximum tip encountered in the analyzed blocks.
+ * @param {bigint} averageTip - average tip encountered in the analyzed blocks.
+ * @param {bigint} medianTip - median (middle value) tip encountered in the analyzed blocks.
+ * @param {bigint} modeTip - mode (most frequent) tip encountered in the analyzed blocks.
+ * @param {bigint} recommendedTip - suggested tip amount (median tip) for optimal inclusion probability.
+ * @param {bigint} p90Tip - 90th percentile tip (90% of tips are below this value).
+ * @param {bigint} p95Tip - 95th percentile tip (95% of tips are below this value).
+ * @param {object} metrics - Optional performance metrics for the analysis.
+ */
+export type TipEstimate = {
```

Comment:
Would it make sense to make the retrieved tips that were used for the estimates also available?

Reply:
e46cd64ad0ed7a7d1067d6f183ad8eba549b313b